src/main/java/org/digma/intellij/plugin/assets/AssetsPanel.java

```diff
@@ -0,0 +1,118 @@
+package org.digma.intellij.plugin.assets;
+
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.editor.colors.impl.AppEditorFontOptions;
+import com.intellij.openapi.editor.colors.impl.FontPreferencesImpl;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.ui.jcef.JBCefBrowser;
+import com.intellij.util.ui.JBUI;
+import org.apache.commons.lang3.StringUtils;
+import org.cef.CefApp;
+import org.cef.browser.CefMessageRouter;
+import org.digma.intellij.plugin.common.JBCefBrowserBuilderCreator;
+import org.digma.intellij.plugin.toolwindow.common.ThemeUtil;
+
+import javax.swing.*;
+import java.awt.*;
+
+import static org.digma.intellij.plugin.ui.list.ListCommonKt.listBackground;
+
+public class AssetsPanel extends JPanel implements Disposable {
+
+    static final String RESOURCE_FOLDER_NAME = "/webview/assets";
+    static final String DOMAIN_NAME = "assets";
+    static final String SCHEMA_NAME = "http";
+
+    private final transient JBCefBrowser jbCefBrowser;
+    private final transient CefMessageRouter cefMessageRouter;
+    private final transient AssetsMessageRouterHandler messageHandler;
+
+    public AssetsPanel(Project project,Disposable parentDisposable) {
+
+        Disposer.register(parentDisposable,this);
+
+        jbCefBrowser = JBCefBrowserBuilderCreator.create()
+                .setUrl("http://" + DOMAIN_NAME + "/index.html")
+                .build();
+        registerAppSchemeHandler(project);
+
+        var jbCefClient = jbCefBrowser.getJBCefClient();
+        cefMessageRouter = CefMessageRouter.create();
+        messageHandler = new AssetsMessageRouterHandler(project,parentDisposable, jbCefBrowser);
+        cefMessageRouter.addHandler(messageHandler, true);
+        jbCefClient.getCefClient().addMessageRouter(cefMessageRouter);
+
+        setLayout(new BorderLayout());
+        setBorder(JBUI.Borders.empty());
+        add(jbCefBrowser.getComponent());
+        setBackground(listBackground());
+
+        addPropertyChangeListener(evt -> {
+            if (project.isDisposed()) return;
+            if ("font".equals(evt.getPropertyName())) {
+                changeFont();
+            }
+        });
+
+        UIManager.addPropertyChangeListener(evt -> {
+            if (project.isDisposed()) return;
+            if ("lookAndFeel".equals(evt.getPropertyName())) {
+                changeTheme();
+            }
+        });
+
+
+        //todo: temporary, this is a very bad way to do it, waiting for help from jetbrains developers
```

Comment:
what is bad? 
what do you think should be the alternative?

Reply:
if i knew a better way i would do it. see that its casting to a concrete class which is not the best because you can't be sure that this is the concrete class always.